### PR TITLE
Ensures threads are stopped in sys_spu_thread_group_join

### PR DESCRIPTION
**Pull request by RipleyTom**
Only sets status to SPU_STATUS_STOPPED_BY_STOP when the thread is actually stopping(end of cpu_task).
sys_spu_thread_group_join will only finish when all threads have actually stopped.

Thanks whym for the logs and elad for helping me figure out the correct way to fix this.

### CODE REVIEW

**Comment by AniLeo** [comment]
Fixes old Catherine regression from earlier this year which makes the game non deterministically freeze after Atlus logo

**Comment by darkdragonz12** [comment]
FIXED FREZZE BLAZE BLUE CENTRAL FICTION!! GREAT WORK 
sin titulo

**Comment by darkdragonz12** [comment]
FIXED FREEZE NARUTO SHIPPUDEN ULTIMATE NINJA STORM 2 
naruto

**Comment by darkdragonz12** [comment]
FIXED  Devil my cry 4 Frezze
devyl my cry 4

**Comment by RainbowCookie32** [comment]
Can confirm, brings Catherine NPUB30698 back ingame for me
capture
Thank you very much for the fix!

**Comment by AniLeo** [comment]
Cutscene issue is unrelated. See #4408

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 1]
```diff
@@ -480,6 +480,21 @@ extern thread_local std::string(*g_tls_log_prefix)();
...
+		// save next PC and current SPU Interrupt status
+		if (!group && offset >= RAW_SPU_BASE_ADDR)
+		{
+			npc = pc | (interrupts_enabled);
+		}
+
+		if (state & cpu_flag::stop)
+		{
+			status |= SPU_STATUS_STOPPED_BY_STOP;
+			if(group) group->cv.notify_one();
```
Comment (kd-11) [inline rpcs3/Emu/Cell/SPUThread.cpp]: missing space after `if`

**Comment by HexyFify** [comment]
Ripley's on a roll! :+1:

**Comment by micaelis72** [comment]
This PR probably fixes all issues on #4875

**Comment by YaPeL** [comment]
does not fix https://github.com/RPCS3/rpcs3/issues/4861 @AniLeo

**Comment by micaelis72** [comment]
Linux tests done!

Indeed fix errors in:

- BLUS30196 [Valkyria Chronicles] Maybe can be considered playable. Only the texture leak on the first battle remains
- BLUS30793 [SSFIVAE] Now goes ingame after the first intro cutscene
- BLJS10184 [Macross 30] No longer freezes on cutscenes or transitions until gameplay. Still shows several errors on the console and when goes ingame, fps drops from 30 to 1-2 fps

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 2]
```diff
@@ -480,6 +480,21 @@ extern thread_local std::string(*g_tls_log_prefix)();
 
 void spu_thread::cpu_task()
 {
+	auto end_cpu_task= [&]()
```
Comment (elad335) [inline rpcs3/Emu/Cell/SPUThread.cpp]:
```suggestion
	auto end_cpu_task = [&]()
```

**Comment by RealMrPascal** [comment]
Build Fixes KH1.5 First Cutscene Blackscreen Freeze and in KH 2.5 After Final Battle Blackscreen Freeze. 

OS: Windows 7
CPU: Intel Core i7 6700K
GPU: Nvidia Geforce GTX 1060 6GB
RAM: 16GB DDR4

**Comment by legend800** [comment]
This PR broke Ridge Racer 7 with SPU LLVM:  https://github.com/RPCS3/rpcs3/issues/5342

**Comment by M-Hassan90s** [comment]
RipleyTom:CleanFix

**Comment by thiagosp89** [comment]
FIFA 13 BLUS30998 and BLES01676 regression. Game stuttering during gameplay. Impossivel to play.

**Pull request merged**